x-pack/test/detection_engine_api_integration/security_and_spaces/group1/export_rules.ts

```diff
@@ -65,6 +77,58 @@ export default ({ getService }: FtrProviderContext): void => {
         expect(bodyToTest).to.eql(getSimpleRuleOutput());
       });
 
+      it('should export a single executed rule with a rule_id', async () => {
+        const ruleId = 'rule-1';
+
+        const rule = await createRule(supertest, log, getSimpleRule(ruleId, true));
+
+        await waitForRuleSuccessOrStatus(supertest, log, rule.id);
+        await waitForEventLogExecuteComplete(es, log, rule.id);
+
+        const { body } = await supertest
+          .post(`${DETECTION_ENGINE_RULES_URL}/_export`)
+          .set('kbn-xsrf', 'true')
+          .send()
+          .expect(200)
+          .parse(binaryToString);
+
+        const bodySplitAndParsed = JSON.parse(body.toString().split(/\n/)[0]);
+
+        expect(bodySplitAndParsed).to.eql({
+          ...getSimpleRuleOutput(ruleId, true),
+          id: rule.id,
+          created_at: rule.created_at,
+          updated_at: rule.updated_at,
```

Comment:
Can we use `expect.objectContaining` here to verify only relevant properties?

Reply:
What's wrong in checking that we export specific fields only? The rule's export format shouldn't change easily. If some new fields appear it must be prominent.

Reply:
We have discussed the drawbacks of using snapshot tests multiple times, and it is worth summarizing them for clarity. Firstly, snapshot tests tend to capture irrelevant information, creating unnecessary noise in the output. Secondly, they hide the intent of the test, making it difficult for developers to determine precisely what matters just by looking at the output. Lastly, snapshot tests are often fragile and require frequent updates, leading to false positives.

To avoid these issues, it's important to make the intent of your test explicit. For instance, if your goal is to verify that dynamic fields do not get exported, it is best to focus on checking that a specific rule was successfully exported and that the output does not contain any dynamic fields. On the other hand, if you want to verify the correctness of the entire exported rule schema, it is better to create a separate test for that purpose. Mixing different cases in one test can lead to confusion and complicate the debugging process.

By explicitly stating the intent of your test and creating separate tests for different cases, you can improve the quality and reliability of your codebase. This approach reduces the likelihood of false positives and makes it easier for developers to understand the test results and debug issues quickly.